### ISSUE TITLE
No error / close callback called when socket is invalidated

### ISSUE THREAD

**Issue by jpcarlino** [gimite/web-socket-js#92]
I'm not completely sure if this is a collateral problem but from the tests i've performed it seems that a flash exception is thrown when the websocket is invalidated before any javascript callback has been called.

My application sends a keepalive packet every 30 seconds to the server. If I suddenly stop the server while connection is in progress, first of all, the client doesn't notice it (no event is triggered by flash bridge). Second, when the heartbeat is sent, flash bridge throws this error:

```
Error calling method on NPObject!
```

the javascript line in question is the send function, as expected:

``` javascript
var result = WebSocket.__flash.send(this.__id, encodeURIComponent(data));
```

looking into flash debugger i see:

``` actionscript
Error: Error #2002: Operation attempted on invalid socket.
at flash.net::Socket/flush()
at com.hurlant.crypto.tls::TLSSocket/flush()
at net.gimite.websocket::WebSocket/send()
at net.gimite.websocket::WebSocketMain/send()
at Function/http://adobe.com/AS3/2006/builtin::apply()
at flash.external::ExternalInterface$/_callIn()
at Function/<anonymous>()
```

so, in short, when socket is invalidated (for example, connection broken) no event is triggered, and it seems i can't catch that error. Any clue?
